app/src/main/java/app/revanced/manager/ui/viewmodel/PatcherViewModel.kt

```diff
@@ -241,20 +289,58 @@ class PatcherViewModel(
     fun open() = installedPackageName?.let(pm::launch)
 
     fun install(installType: InstallType) = viewModelScope.launch {
+        var isInstallingRegularly = false
         try {
             isInstalling = true
+
+            val currentPackageInfo = pm.getPackageInfo(outputFile)
+                ?: throw Exception("Failed to load application info")
+
+            // If the app is currently installed
+            val existingPackageInfo = pm.getPackageInfo(packageName)
+            if (existingPackageInfo != null) {
+                // Check if the app version is less than the installed version
+                if (pm.getVersionCode(currentPackageInfo) < pm.getVersionCode(existingPackageInfo)) {
+                    // Exit if the selected app version is less than the installed version
+                    installerStatusDialogModel.packageInstallerStatus = PackageInstaller.STATUS_FAILURE_CONFLICT
+                    return@launch
+                }
+            }
+
             when (installType) {
                 InstallType.DEFAULT -> {
+                    // Check if the app is mounted as root
+                    // If it is, unmount it first, silently
+                    if (rootInstaller.hasRootAccess() && rootInstaller.isAppMounted(packageName)) {
+                        rootInstaller.unmount(packageName)
+                    }
+
+                    // Install regularly
+                    isInstallingRegularly = true
```

Comment:
Move this after `installApp` so it is unset if `installApp` throws an exception for some reason